src/app/common/utils/copy-to-clipboard.ts

```diff
@@ -0,0 +1,6 @@
+import toast from 'react-hot-toast';
```

Comment:
It does look like `writeText()` is a promise and needs to be handled or will get issues?

Reply:
Thanks. I can add an `await` just in case. Its working now but we can see what Kyran says. 

I don't think it needs to be a hook if it works with a simple callback

Reply:
I copied from elsewhere. 

Noticed the execCommand is deprecated, does it work the newer way with Clipboard API?

<img width="1089" alt="image" src="https://github.com/leather-wallet/extension/assets/1618764/71b615d4-5854-412d-9224-3bda29c278fb">

Reply:
I saw that deprecation warning also. Using the `execCommand` works in other areas of the APP but not when I try and do it inside of a Radix Dialog. 

I checked and if I update `useClipboard` to use `navigator.clipboard.writeText` it will work so I can refactor `useClipboard` to use `writeText` instead and avoid adding a new function if you think that's best. 

Looking at how we use it and  I don't think it needs to be a react hook and sometimes that just adds complexity. 

For example, if I am using the hook in the current `ReceiveModal` I need to use it 3 times for each address:
```
  const { onCopy: onCopyBtc } = useClipboard(btcAddressNativeSegwit);
  const { onCopy: onCopyStx } = useClipboard(stxAddress);
  const { onCopy: onCopyOrdinal } = useClipboard(btcAddressTaproot);
```

We also then have this `copyToClipboard(` function repeated a lot that calls the copy and tracks analytics:
```
  function copyToClipboard(copyHandler: () => void, tracker = 'copy_address_to_clipboard') {
    void analytics.track(tracker);
    toast.success('Copied to clipboard!');
    copyHandler();
  }
```

Changing it to be a standard function and components just need to receive text (e.g. an address) and pass it to the function copy function. 

A potential issue is the analytics tracking callback but I should be able to add that to the new function

Reply:
I just added an `await` here for now but opened a new task to investigate refactoring the hook properly: https://github.com/leather-wallet/extension/issues/5017